spacy_transformers/pipeline_component.py

```diff
@@ -212,7 +212,7 @@ def pipe(self, stream: Iterable[Doc], *, batch_size: int = 128) -> Iterator[Doc]
                 self.set_annotations(subbatch, self.predict(subbatch))
             yield from outer_batch
 
-    def predict(self, docs: Iterable[Doc]) -> FullTransformerBatch:
+    def predict(self, docs: List[Doc]) -> FullTransformerBatch:
```

Comment:
Why are these redefined? The documentation & `TrainablePipe` contract do specify `Iterable` for `predict` and `set_annotations`. Can we instead make sure the functions do in fact work for any `Iterable`?

Reply:
I think that `Iterable` is kind of a mistake because I feel like a batch should really be at least `Sized` and everywhere we have typed most of our models as something like `Model[List[Doc], Floats2d]`, so what has happened is that we're ignoring the `Model.predict` type errors instead of making it consistent (and we don't notice most of them in cython; if you pass an unsized `Iterable` to `Tagger` it will crash).

I think it's a little weird that the default `predict` for any of our model definitions would look like:

```
def predict(self, docs: Iterable[Doc]):
    self.model.predict(list(docs))
```

I think `Iterable[Doc]` makes sense for `pipe`, but less so for `predict` and `set_annotations`. Or we would need to refactor all the model definitions and existing pipes to actually work with `Iterable`.

But I can make `Iterable` work here for now.